src/components/utils/ListData.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from 'react';
+import { List } from 'antd';
+import Router, { useRouter } from 'next/router';
+import { isEmpty } from 'lodash';
+import Section from './Section';
+
+type Props = {
+  className?: string,
+  dataSource: any,
+  renderItem: (item: any, index: number) => JSX.Element
+  title?: React.ReactNode
+};
+
+export default (props: Props) => {
+  const { dataSource,renderItem, className, title } = props;
+  const total = dataSource.length;
+  const DEFAULT_PAGE_SIZE = 1;
+  const DEFAULT_CURENT_PAGE = 1;
+
+  const router = useRouter();
+  const routerQuery = router.query;
+
+  const [ currentPage, setCurrentPage ] = useState(DEFAULT_CURENT_PAGE);
+  const [ pageSize, setPageSize ] = useState(DEFAULT_PAGE_SIZE);
+
+  useEffect(() => {
+    if (isEmpty(routerQuery)) {
+      setPageSize(DEFAULT_PAGE_SIZE);
+      setCurrentPage(DEFAULT_CURENT_PAGE);
+
+      routerQuery.size = DEFAULT_PAGE_SIZE.toString();
+      routerQuery.page = DEFAULT_CURENT_PAGE.toString();
+
+      Router.push({
+        pathname: router.pathname,
+        query: routerQuery
+      }).catch(console.log);
+    } else {
+      const page = parseInt(routerQuery.page as string, 10);
+      const _pageSize = parseInt(routerQuery.size as string, 10);
+
+      setCurrentPage(page > 0 ? page : DEFAULT_PAGE_SIZE);
+      setPageSize(_pageSize > 0 && _pageSize < 100 ? _pageSize : DEFAULT_PAGE_SIZE);
```

Comment:
Create a config constant `const MAX_PAGE_SIZE = 100` and use it here:
```suggestion
      setPageSize(_pageSize > 0 && _pageSize <= MAX_PAGE_SIZE ? _pageSize : DEFAULT_PAGE_SIZE);
```